src/main/scala/GameHelpers.scala

```diff
@@ -0,0 +1,69 @@
+import scala.util.Random
+
+object GameHelpers {
+
+  def fillRack(rack: List[Tile], letterBag: List[Tile]): (List[Tile], List[Tile]) = {
+    val (drawnTiles, newBag) = letterBag.splitAt(7 - rack.length)
+    val filledRack = rack ++ drawnTiles
+
+    if (filledRack.length < 7) {
+      println("\u001b[0m---------")
+      print("\u001b[31;1mWARNING: ")
+      println("\u001b[36mThe letter bag is empty.")
+      print("\u001b[0m")
+    }
+
+    (filledRack, newBag)
+  }
+
+  def removeTilesFromRack(word: String, rack: List[Tile]): List[Tile] = {
+    val wordTiles: List[Tile] = word.toList.map(c => Tile(c))
+    rack diff wordTiles
+  }
+
+  def calculateWordScore(word: String): Int = {
+    word.toList.map(Tile(_).value).sum
+  }
+
+  def isInDictionary(word: String): Boolean = {
+    if (Dictionary.all.contains(word.toLowerCase())) true else false
+  }
+
+  def isInRack(word: String, rack: List[Tile]): Boolean = {
+    val sortedWord = word.toList.sortWith(_.compareTo(_) < 0)
+    val sortedRack = rack.map(_.letter).sortWith(_.compareTo(_) < 0)
+
+    def loop(wL: List[Char], wI: Int, rL: List[Char], rI: Int): Boolean = {
```

Comment:
It's great that you understand recursion enough to write this! Bravo! Another way to write this might be:

```scala
import scala.annotation.tailrec

@tailrec
def isInRack(word: String, rack: List[Tile]): Boolean = {
    go(word.toList, rack.map(_.c))

    @tailrec
    def go(wordLetters: List[Char], rackLetters: List[Char]): Boolean =
        wordLetters match {
            case Nil => true
            case h :: t => rackLetters.contains(h) && go(t, rackLetters diff List(h))
        }
}
```